enhancements/windows-containers/container-runtime-containerd.md

```diff
@@ -0,0 +1,205 @@
+If the MachineSet uses a VM image that comes with Docker, containerd will run along with Docker service. But Kubernetes will
+use only containerd. Customers can stop Docker service if it is no longer needed. Even if Docker gets installed at any time,
+it can co-exist with containerd.
+
+Steps to install containerd as service:
+- scp containerd/related executables into Windows VM
+- copy the files into `C:\k\bin` folder
+- create containerd config file
+- register containerd as service
+- start containerd as a Windows service
+
+## Network changes
+Current CNI/IPAM will be used for containerd and no change in HNS-Network and HNS-Endpoint creation
+steps. The config file which will be used by containerd will point to the same CNI/IPAM executables.
+
+## WMCO CLI option
+WMCO CLI option 'runtime' will be used to make containerd as the default runtime.  In the 4.10 community operator, it
+will be specified as `--runtime containerd`. In the absence of this flag, Docker will become the default runtime. We remove
+this flag in 4.11 and make containerd the default runtime. At any given point in time, we have one runtime support and
+do not allow switching between runtime.
+
+## Containerd logging
+Containerd can be started with parameters in which we can enable logging and specify the file path to
+log warnings/errors. Log files will be stored at c:\var\log\containerd. Pod logs will be stored in the
+same location (C:\var\log\pods)
+
+## Upgrade
+WMCO supports upgrade. Please refer [WMCO-Upgrade](https://github.com/openshift/enhancements/blob/master/enhancements/windows-containers/windows-machine-config-operator-upgrades.md)
+enhancement proposal document to learn more about WMCO upgrade.
+
+The procedure for an upgrade is as follows:
+1) MachineSet
+   * As part of upgrade Machine object gets deleted, resulting in the drain and deletion of the Windows node.
+   * New Windows VM will be created.
+   * The upgraded WMCO instance will configure Windows VM with containerd as the default runtime.
+   * Once the Windows node joins the Kubernetes cluster, the user can deploy pods on the Windows node.
+2) BYOH ( Bring Your Own Host)
+   * upgrade process will uninstall kubelet, kube-proxy, CNI, and hybrid-overlay components that were installed by WMCO.
+   * Windows OS-specific configurations like HNS-Network that were created by WMCO instance would be deleted.
+   * Once cleanup is done, WMCO instance will install containerd along with kubelet, kube-proxy, CNI and
+     hybrid-overlay.
+   * Kuebelet will start as a service. Containerd will be the default runtime.
+   * Once the Windows node joins the Kubernetes cluster, the user can deploy pods on the Windows node.
+
+### Risks and Mitigations
+
+* If the cluster is upgraded, and the new version introduces an issue due to containerd
+  that should be addressed because Docker runtime support won't be available in kubelet.
```

Comment:
Please state in what way this should be addressed

for example within WMCO and/or a patch to containerd

Reply:
These are mainly containerd specific issues. I moved statements around to address the mitigation.